From 1183902c899dbf58bcba8324b7d993b532723cfd Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Mon, 26 Dec 2016 09:31:04 +0100
Subject: [PATCH 0681/1345] telephony: mvebu: tdm2c: initial cleanup

commit  bc1cc5a259dc6ae3bc0d989c233278bb9869d0e0 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

* Move defines and tdm2c_ch_info structure to tdm2c.h
* Remove unused get_tdm_base() function

Change-Id: Ibf28427c48844c53fe4e4d476a1550b08b40c520
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/35050
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/telephony/mvebu_phone/tdm2c/tdm2c.c |   26 -------------------------
 drivers/telephony/mvebu_phone/tdm2c/tdm2c.h |   28 +++++++++++++++++++++++++--
 2 files changed, 26 insertions(+), 28 deletions(-)

diff --git a/drivers/telephony/mvebu_phone/tdm2c/tdm2c.c b/drivers/telephony/mvebu_phone/tdm2c/tdm2c.c
index 6ea6e5a..20e7255 100644
--- a/drivers/telephony/mvebu_phone/tdm2c/tdm2c.c
+++ b/drivers/telephony/mvebu_phone/tdm2c/tdm2c.c
@@ -96,27 +96,6 @@
 
 #include "mv_phone.h"
 
-/* Defines */
-#define INT_SAMPLE			2
-#define BUFF_IS_FULL			1
-#define BUFF_IS_EMPTY			0
-#define FIRST_INT			1
-#define TOTAL_BUFFERS			2
-#define MV_TDM_NEXT_BUFFER(buf)		((buf + 1) % TOTAL_BUFFERS)
-#define MV_TDM_PREV_BUFFER(buf, step)	((TOTAL_BUFFERS + buf - step) % TOTAL_BUFFERS)
-#define MV_TDM_CS			0
-#define BUFF_INVALID			-1
-
-/* TDM channel info structure */
-struct tdm2c_ch_info {
-	u8 ch;
-	u8 *rxBuffVirt[TOTAL_BUFFERS], *txBuffVirt[TOTAL_BUFFERS];
-	dma_addr_t rxBuffPhys[TOTAL_BUFFERS], txBuffPhys[TOTAL_BUFFERS];
-	u8 rxBuffFull[TOTAL_BUFFERS], txBuffFull[TOTAL_BUFFERS];
-	u8 rxCurrBuff, txCurrBuff;
-	u8 rxFirst;
-};
-
 /* Globals */
 static u8 *rx_aggr_buff_virt, *tx_aggr_buff_virt;
 static u8 rx_int, tx_int;
@@ -379,11 +358,6 @@ static void tdm2c_reset(void)
 	}
 }
 
-void __iomem *get_tdm_base(void)
-{
-	return regs;
-}
-
 int tdm2c_init(void __iomem *base, struct device *dev,
 	       struct mv_phone_params *tdmParams, struct mv_phone_data *halData)
 {
diff --git a/drivers/telephony/mvebu_phone/tdm2c/tdm2c.h b/drivers/telephony/mvebu_phone/tdm2c/tdm2c.h
index 2ab135f..188b652 100644
--- a/drivers/telephony/mvebu_phone/tdm2c/tdm2c.h
+++ b/drivers/telephony/mvebu_phone/tdm2c/tdm2c.h
@@ -315,9 +315,33 @@
 	((SAMPLES_BUFF_SIZE(bandMode, factor)<<TOTAL_CNT_OFFS) |	\
 	(INT_SAMPLE<<INT_CNT_OFFS))
 
-/* APIs */
-void __iomem *get_tdm_base(void);
+/* Defines */
+#define INT_SAMPLE			2
+#define BUFF_IS_FULL			1
+#define BUFF_IS_EMPTY			0
+#define FIRST_INT			1
+#define TOTAL_BUFFERS			2
+#define MV_TDM_NEXT_BUFFER(buf)		((buf + 1) % TOTAL_BUFFERS)
+#define MV_TDM_PREV_BUFFER(buf, step)	((TOTAL_BUFFERS + buf - step) % TOTAL_BUFFERS)
+#define MV_TDM_CS			0
+#define BUFF_INVALID			-1
 
+/* Structures */
+/* TDM channel info structure */
+struct tdm2c_ch_info {
+	u8 ch;
+	u8 *rxBuffVirt[TOTAL_BUFFERS];
+	u8 *txBuffVirt[TOTAL_BUFFERS];
+	dma_addr_t rxBuffPhys[TOTAL_BUFFERS];
+	dma_addr_t txBuffPhys[TOTAL_BUFFERS];
+	u8 rxBuffFull[TOTAL_BUFFERS];
+	u8 txBuffFull[TOTAL_BUFFERS];
+	u8 rxCurrBuff;
+	u8 txCurrBuff;
+	u8 rxFirst;
+};
+
+/* APIs */
 void tdm2c_release(void);
 int tdm2c_pcm_stop_int_miss(void);
 void tdm2c_pcm_start(void);
-- 
1.7.9.5

